### DISCUSSION TITLE
Simple way to create a RequestQueryParamSerializer for all DateTimes?

### DISCUSSION THREAD

**Discussion by VictorioBerra** [canton7/RestEase#190] [Q&A] [answered]
I created my own `RequestQueryParamSerializer` but I find I have to manually add to all `DateTimes` via:

```csharp
public interface ISomeApi
{
	[Get("foo")]
	Task FooAsync([Query(QuerySerializationMethod.Serialized)] DateTimeOffset param);
}
```

If I use `[SerializationMethods(Query = QuerySerializationMethod.Serialized)]` on the interface to force all query params to use it, I would need to somehow check for DateTime only and then defer to the default serializer.

I do not see a way to call `base.SerializeQueryParam` or `base.SerializeQueryCollectionParam`.

**Comment by canton7** [comment, marked as answer]
That is indeed how things are. If you want to defer to another serializer, instantiate that serializer in your serializer.

Don't forget that format strings are supported, if that covers your use-case.
